app/plugins/Wallet/js/lifecycle.js

```diff
@@ -1,89 +1,163 @@
 	if (wallet.unlocked && wallet.encrypted) {
+		eID('confirmed').style.display = 'inline';
+		eID('unconfirmed').style.display = 'inline';
 		eID('confirmed').innerHTML = 'Balance: ' + bal + ' S';
-		eID('uncomfirmed').innerHTML = 'Pending: ' + pend + ' S';
-		eID('confirmed').style.removeProperty('display');
-		eID('uncomfirmed').style.removeProperty('display');
+		eID('unconfirmed').innerHTML = 'Pending: ' + pend + ' S';
 	} else {
-		eID('confirmed').style.setProperty('display', 'none');
-		eID('uncomfirmed').style.display = 'none';
+		eID('confirmed').style.display = 'none';
+		eID('unconfirmed').style.display = 'none';
 	}
 });
 
-/*
-// Adds an address to the address list
+// Append wallet address to address list
+function appendAddress(addr) {
+	// Create only new addresses
+	if (typeof(addr) == 'undefined') { return; }
+	var addrElement = eID('addressbp').cloneNode(true);
+
+	// DOM shortcut
+	var field = function(selector) {
+		return addrElement.querySelector(selector);
+	};
+
+	// Insert values
+	field('.listnum').innerHTML = eID('address-list').childNodes.length + 1;
+	field('.address').innerHTML = addr.address;
+	field('.address').id = addr.address;
+	field('.address').addEventListener("click", getAddress);
+
+	// Display address
+	eID('address-list').appendChild(addrElement);
+	show(addrElement);
+	return;
+}
+function getAddress(event) {
+	updateAddrTxn(event.target.id);
+}
+
 function appendTransaction(txn) {
-	if (eID(txn.transactionid)) {
-		return;
+	if (typeof(txn) == 'undefined') { return; }
+
+	// Add only new transactions
+	if (eID(txn.transactionid)) { return; }
+	var txnElement = eID('transactionbp').cloneNode(true);
+	txnElement.id = txn.transactionid;
+	txnElement.timestamp = txn.confirmationtimestamp * 1000;
+
+	// DOM shortcut
+	var field = function(selector) {
+		return txnElement.querySelector(selector);
+	};
+
+	// Compute transaction net amount
+	var amount = new BigNumber(0);
+	if (txn.inputs) {
+		txn.inputs.forEach( function(input) {
+			if (input.walletaddress) {
+				amount = amount.sub(input.value);
+			}
+		});
+	}
+	if (txn.outputs) {
+		txn.outputs.forEach( function(output) {
+			if (output.walletaddress) {
+				amount = amount.add(output.value);
+			}
+		});
 	}
-	var entry = eID('transactionbp').cloneNode(true);
-	entry.id = txn.transactionid;
+	var ConversionFactor = new BigNumber(10).pow(24);
+	amount = amount.dividedBy(ConversionFactor).round(2);
 
-	// Determine how to display transaction
-	// Have to use !== logic to represent miner payouts
-	var sign, unit, amount, related;
-	var ft = txn.fundtype.split(' ');
-	sign = ft[1] !== 'input' ? '+' : '-';
-	amount = ft[0] !== 'siafund' ? convertSiacoin(txn.value) : txn.value;
-	unit = ft[0] !== 'siafund' ? ' Siacoin ' : ' Siafund ';
-	related = ft[1] !== 'input' ? ' received into ' : ' sent from ';
-	related += txn.relatedaddress;
+	// Format transaction timestamp
+	var timestamp = new Date(txn.confirmationtimestamp * 1000);
```

Comment:
You do this timestamp * 1000 multiplication twice. Perhaps make a helper function for it?